raster/src/main/scala/geotrellis/raster/summary/polygonal/visitors/TileFoldingVisitor.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package geotrellis.raster.summary.polygonal.visitors
+
+import geotrellis.raster._
+import geotrellis.raster.summary.CellVisitor
+
+abstract class TileFoldingVisitor
```

Comment:
So that brings up a question. Should this be public? We could make it private if we don't want to support it.

Reply:
But otherwise yes, this needs some docs. Planning to add those for everything if we're ok with the general look/feel of the API changes here.

Reply:
I'm fine with making it `private`. It's a little defensive but there is not so much and such useful logic in there that it can't be easily replaced in place by user.

Reply:
Yeah that's what I went with actually. It's fairly specific actually. The only thing you could really use it for is stddev and a few other basic aggregation functions. Which if someone wants we should probably implement here like the others.